# Libraries and Data Import

In [1]:
import json
import pandas as pd

In [2]:
with open('stations_data.json', 'r') as f:
    raw_data = json.load(f)

df_data = pd.DataFrame(raw_data)
len(raw_data)
df_data.shape

(19716, 31)

# Data Analysis

In [7]:
df_data.keys()

Index(['DataProvider', 'OperatorInfo', 'UsageType', 'StatusType',
       'SubmissionStatus', 'UserComments', 'PercentageSimilarity',
       'MediaItems', 'IsRecentlyVerified', 'DateLastVerified', 'ID', 'UUID',
       'ParentChargePointID', 'DataProviderID', 'DataProvidersReference',
       'OperatorID', 'OperatorsReference', 'UsageTypeID', 'UsageCost',
       'AddressInfo', 'Connections', 'NumberOfPoints', 'GeneralComments',
       'DatePlanned', 'DateLastConfirmed', 'StatusTypeID',
       'DateLastStatusUpdate', 'MetadataValues', 'DataQualityLevel',
       'DateCreated', 'SubmissionStatusTypeID'],
      dtype='str')

In [3]:
df_data

,DataProvider,OperatorInfo,UsageType,StatusType,SubmissionStatus,UserComments,PercentageSimilarity,MediaItems,IsRecentlyVerified,DateLastVerified,...,NumberOfPoints,GeneralComments,DatePlanned,DateLastConfirmed,StatusTypeID,DateLastStatusUpdate,MetadataValues,DataQualityLevel,DateCreated,SubmissionStatusTypeID
0,"{'WebsiteURL': 'http://openchargemap.org', 'Co...",{'WebsiteURL': 'https://www.iberdrola.es/clien...,"{'IsPayAtLocation': False, 'IsMembershipRequir...","{'IsOperational': True, 'IsUserSelectable': Tr...","{'IsLive': True, 'ID': 200, 'Title': 'Submissi...",None,None,None,True,2026-03-04T23:03:00Z,...,4.0,NaN,None,None,50,2026-03-04T23:03:00Z,None,1,2026-03-04T23:03:00Z,200
1,"{'WebsiteURL': 'http://openchargemap.org', 'Co...",{'WebsiteURL': 'https://www.iberdrola.es/clien...,"{'IsPayAtLocation': False, 'IsMembershipRequir...","{'IsOperational': True, 'IsUserSelectable': Tr...","{'IsLive': True, 'ID': 200, 'Title': 'Submissi...",None,None,None,True,2026-03-04T23:02:00Z,...,6.0,NaN,None,None,50,2026-03-04T23:02:00Z,None,1,2026-03-04T23:02:00Z,200
2,"{'WebsiteURL': 'http://openchargemap.org', 'Co...","{'WebsiteURL': None, 'Comments': 'For use when...","{'IsPayAtLocation': False, 'IsMembershipRequir...","{'IsOperational': True, 'IsUserSelectable': Tr...","{'IsLive': True, 'ID': 200, 'Title': 'Submissi...",None,None,None,True,2026-03-04T22:46:00Z,...,1.0,NaN,None,None,50,2026-03-04T22:46:00Z,None,1,2026-03-04T22:46:00Z,200
3,"{'WebsiteURL': 'http://openchargemap.org', 'Co...",{'WebsiteURL': 'https://www.repsol.es/particul...,"{'IsPayAtLocation': None, 'IsMembershipRequire...","{'IsOperational': True, 'IsUserSelectable': Tr...","{'IsLive': True, 'ID': 200, 'Title': 'Submissi...",None,None,None,True,2026-03-01T17:31:00Z,...,2.0,NaN,None,None,50,2026-03-01T17:31:00Z,None,1,2026-03-01T17:31:00Z,200
4,"{'WebsiteURL': 'http://openchargemap.org', 'Co...",{'WebsiteURL': 'https://www.aparcamentsbsm.cat...,"{'IsPayAtLocation': None, 'IsMembershipRequire...","{'IsOperational': True, 'IsUserSelectable': Tr...","{'IsLive': True, 'ID': 200, 'Title': 'Submissi...",None,None,None,True,2026-03-01T17:03:00Z,...,4.0,NaN,None,None,50,2026-03-01T17:03:00Z,None,1,2026-03-01T17:03:00Z,200
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
19711,"{'WebsiteURL': 'http://openchargemap.org', 'Co...",{'WebsiteURL': 'https://www.aparcamentsbsm.cat...,"{'IsPayAtLocation': False, 'IsMembershipRequir...","{'IsOperational': True, 'IsUserSelectable': Tr...","{'IsLive': True, 'ID': 200, 'Title': 'Submissi...",None,None,None,False,2022-10-21T22:32:00Z,...,1.0,NaN,None,None,50,2022-10-21T22:32:00Z,None,1,2014-06-03T05:17:00Z,200
19712,"{'WebsiteURL': 'http://openchargemap.org', 'Co...","{'WebsiteURL': 'http://www.edp.pt/', 'Comments...","{'IsPayAtLocation': False, 'IsMembershipRequir...","{'IsOperational': True, 'IsUserSelectable': Tr...","{'IsLive': True, 'ID': 200, 'Title': 'Submissi...","[{'ID': 4254, 'ChargePointID': 23788, 'Comment...",None,None,True,2026-01-30T13:35:00Z,...,12.0,NaN,None,None,50,2026-01-30T13:35:00Z,None,1,2014-05-29T18:23:00Z,200
19713,"{'WebsiteURL': 'http://openchargemap.org', 'Co...","{'WebsiteURL': 'https://www.barterenergy.es/',...","{'IsPayAtLocation': False, 'IsMembershipRequir...","{'IsOperational': True, 'IsUserSelectable': Tr...","{'IsLive': True, 'ID': 200, 'Title': 'Submissi...",None,None,"[{'ID': 311, 'ChargePointID': 23085, 'ItemURL'...",False,2024-07-06T15:53:00Z,...,6.0,NaN,None,None,50,2024-07-06T15:53:00Z,None,1,2014-04-11T07:24:00Z,200
19714,"{'WebsiteURL': 'http://openchargemap.org', 'Co...",None,"{'IsPayAtLocation': False, 'IsMembershipRequir...","{'IsOperational': True, 'IsUserSelectable': Tr...","{'IsLive': True, 'ID': 200, 'Title': 'Submissi...",None,None,None,False,2016-04-24T19:52:00Z,...,3.0,NaN,None,None,50,2016-04-24T19:52:00Z,None,1,2011-11-24T18:40:00Z,200


## Main operators

In [4]:
print(df_data["OperatorID"].value_counts().sort_values(ascending=False).to_string())

OperatorID
45.0      3684
2247.0    2501
207.0     1769
91.0      1548
3499.0    1184
1.0       1062
3377.0     753
3401.0     669
3459.0     562
3276.0     503
3324.0     482
3283.0     459
3371.0     382
3583.0     354
3556.0     331
3571.0     276
3750.0     218
3557.0     206
38.0       198
3464.0     169
3550.0     147
3279.0     134
3584.0     128
3798.0     121
3305.0     121
3588.0     114
3577.0     107
3559.0     107
3275.0     106
3534.0     104
3786.0      83
103.0       77
3960.0      74
3552.0      66
3945.0      63
3956.0      62
3975.0      58
3930.0      50
3299.0      50
200.0       44
3558.0      40
3572.0      33
3785.0      32
3955.0      31
3523.0      29
3809.0      27
3608.0      27
3949.0      24
3632.0      24
3489.0      23
3816.0      23
3553.0      22
3889.0      22
3671.0      19
3586.0      18
3880.0      18
3704.0      17
3312.0      15
3770.0      14
3947.0      13
55.0        12
210.0       11
3772.0      10
23.0        10
3278.0       9
212.0        9

In [10]:
op_45 = df_data.loc[df_data["OperatorID"] == 45].head(1)["OperatorInfo"]
op_45

2    {'WebsiteURL': None, 'Comments': 'For use when...
Name: OperatorInfo, dtype: object

In [11]:
op_2247 = df_data.loc[df_data["OperatorID"] == 2247].head(1)["OperatorInfo"]
op_2247

0    {'WebsiteURL': 'https://www.iberdrola.es/clien...
Name: OperatorInfo, dtype: object

In [12]:
op_207 = df_data.loc[df_data["OperatorID"] == 207].head(1)["OperatorInfo"]
op_207

66    {'WebsiteURL': 'http://www.endesa.es/', 'Comme...
Name: OperatorInfo, dtype: object

In [13]:
op_91 = df_data.loc[df_data["OperatorID"] == 91].head(1)["OperatorInfo"]
op_91

3    {'WebsiteURL': 'https://www.repsol.es/particul...
Name: OperatorInfo, dtype: object

In [14]:
op_3499 = df_data.loc[df_data["OperatorID"] == 3499].head(1)["OperatorInfo"]
op_3499

163    {'WebsiteURL': '  https://www.mercadona.es/ ',...
Name: OperatorInfo, dtype: object

In [15]:
op_1 = df_data.loc[df_data["OperatorID"] == 1].head(1)["OperatorInfo"]
op_1

34    {'WebsiteURL': None, 'Comments': None, 'PhoneP...
Name: OperatorInfo, dtype: object

In [16]:
op_3377 = df_data.loc[df_data["OperatorID"] == 3377].head(1)["OperatorInfo"]
op_3377

8    {'WebsiteURL': 'http://www.etecnic.es', 'Comme...
Name: OperatorInfo, dtype: object

### Summary of information about main Operators

* Operator 45 (3684 chargers): 'Comments': 'For use when the operator of the equipment is a single business owner connected to the location and equipment is not part of a larger network', 'ID': 45, 'Title': '(Business Owner at Location)'

* Operator 2247 (2501 chargers): Iberdrola

* Operator 207 (1769 chargers): Endesa

* Operator 91 (1548 chargers): Repsol

* Operator 3499 (1184 chargers): Mercadona

* Operator 1 (1062 chargers): Unknown Operator

* Operator 3377 (753 chargers): Etecnic

## Columns Analysis

https://openchargemap.org/develop/api#/schemas/POI

Columns found in the data:
- DataProvider: Source of the data. Not relevant for our system.
- **OperatorInfo: Information about the operator of the charging station. Relevant**
- UsageType: Type of usage (e.g., public, private). We assumme all are selectable. Not relevant for our system.
- **StatusType: Operational status of the station. We discard those that are not operational (e.g., "Disabled", "Inactive")**
- SubmissionStatus: Status of the data submission. Not relevant for our system.
- UserComments: User comments about the station. Not relevant for our system.
- PercentageSimilarity: Null value. Not relevant for our system.
- MediaItems: Media items related to the station. Not relevant for our system.
- IsRecentlyVerified: Whether the station has been recently verified. Not relevant for our system.
- DateLastVerified: Date of the last verification. Not relevant for our system.
- **ID: Unique identifier for the station. Relevant.**
- UUID: Unique identifier for the station. Not relevant for our system.
- ParentChargePointID: Null value. Not relevant for our system.
- DataProviderID: Source of the data. Not relevant for our system.
- DataProvidersReference: Identifier for the operator. Not relevant for our system.
- OperatorID: Identifier for the operator. Not relevant for our system.
- OperatorsReference: Identifier for the operator. Not relevant for our system.
- UsageTypeID: Reference for the type of usage. Not relevant for our system.
- **UsageCost: Cost of using the station. Relevant (if available)**
- **AddressInfo: Information about the station's location. Relevant**
- **Connections: Information about the station's different connection types. Relevant**
- **NumberOfPoints: Number of charging points available at the station. Relevant**
- GeneralComments: General comments about the station. Not relevant for our system.
- DatePlanned: Null value. Not relevant for our system.
- DateLastConfirmed: Null value. Not relevant for our system.
- StatusTypeID: Reference for the operational status. Not relevant for our system.
- DateLastStatusUpdated: Date of the last status update. Not relevant for our system.
- MetadataValues: Null value (>99% of rows). Not relevant for our system.
- DataQualityLevel: Unused value. Not relevant for our system.
- DateCreated: Date of creation of the station entry. Not relevant for our system.
- SubmissionStatusTypeID: Reference for the submission status. Not relevant for our system.

In [17]:
clean_data = df_data.copy()

clean_data = clean_data[["ID", "OperatorInfo", "StatusType", "UsageCost", "AddressInfo", "Connections", "NumberOfPoints"]]

In [18]:
clean_data

,ID,OperatorInfo,StatusType,UsageCost,AddressInfo,Connections,NumberOfPoints
0,479753,{'WebsiteURL': 'https://www.iberdrola.es/clien...,"{'IsOperational': True, 'IsUserSelectable': Tr...","0,45€/kWh","{'ID': 480142, 'Title': 'Parking Puerto Portal...","[{'ID': 805482, 'ConnectionTypeID': 33, 'Conne...",4.0
1,479752,{'WebsiteURL': 'https://www.iberdrola.es/clien...,"{'IsOperational': True, 'IsUserSelectable': Tr...","0,45€/kWh","{'ID': 480141, 'Title': 'Puerto Portals. Capit...","[{'ID': 805481, 'ConnectionTypeID': 33, 'Conne...",6.0
2,479751,"{'WebsiteURL': None, 'Comments': 'For use when...","{'IsOperational': True, 'IsUserSelectable': Tr...",,"{'ID': 480140, 'Title': 'BP Guaza', 'AddressLi...","[{'ID': 805478, 'ConnectionTypeID': 33, 'Conne...",1.0
3,479605,{'WebsiteURL': 'https://www.repsol.es/particul...,"{'IsOperational': True, 'IsUserSelectable': Tr...",NaN,"{'ID': 479994, 'Title': 'Seur Manzanares', 'Ad...","[{'ID': 805260, 'ConnectionTypeID': 25, 'Conne...",2.0
4,479604,{'WebsiteURL': 'https://www.aparcamentsbsm.cat...,"{'IsOperational': True, 'IsUserSelectable': Tr...",Free,"{'ID': 479993, 'Title': 'B:SM Avinguda Diagona...","[{'ID': 805259, 'ConnectionTypeID': 28, 'Conne...",4.0
...,...,...,...,...,...,...,...
19711,23803,{'WebsiteURL': 'https://www.aparcamentsbsm.cat...,"{'IsOperational': True, 'IsUserSelectable': Tr...",Free,"{'ID': 24148, 'Title': 'B:SM 52 - Tarradelles-...","[{'ID': 25886, 'ConnectionTypeID': 28, 'Connec...",1.0
19712,23788,"{'WebsiteURL': 'http://www.edp.pt/', 'Comments...","{'IsOperational': True, 'IsUserSelectable': Tr...","parking fee + 0,20€/kWh (0,01€/min parked with...","{'ID': 24133, 'Title': 'Parking APK2 Mondragon...","[{'ID': 25817, 'ConnectionTypeID': 25, 'Connec...",12.0
19713,23085,"{'WebsiteURL': 'https://www.barterenergy.es/',...","{'IsOperational': True, 'IsUserSelectable': Tr...",NaN,"{'ID': 23430, 'Title': 'Hotel Santos Las Arena...","[{'ID': 24788, 'ConnectionTypeID': 25, 'Connec...",6.0
19714,8784,None,"{'IsOperational': True, 'IsUserSelectable': Tr...",Paying small fee with Prepaid RFID card + cost...,"{'ID': 8681, 'Title': 'Parking Plaza de Cánova...","[{'ID': 89854, 'ConnectionTypeID': 28, 'Connec...",3.0


In [19]:
clean_data.keys()

Index(['ID', 'OperatorInfo', 'StatusType', 'UsageCost', 'AddressInfo',
       'Connections', 'NumberOfPoints'],
      dtype='str')

# Data Cleaning

In [20]:
operators_dict = {
    2247: "Iberdrola",
    207: "Endesa",
    91: "Repsol",
    3499: "Mercadona",
    3377: "Etecnic",
}

In [21]:
clean_data = clean_data[
    clean_data["StatusType"].apply(lambda x: isinstance(x, dict) and x.get("Title") == "Operational")
]

print("Not operational:", len(df_data) - len(clean_data))

Not operational: 1382


In [22]:
print(clean_data["AddressInfo"].iloc[0].keys())

dict_keys(['ID', 'Title', 'AddressLine1', 'AddressLine2', 'Town', 'StateOrProvince', 'Postcode', 'CountryID', 'Country', 'Latitude', 'Longitude', 'ContactTelephone1', 'ContactTelephone2', 'ContactEmail', 'AccessComments', 'RelatedURL', 'Distance', 'DistanceUnit'])


In [23]:
data_to_insert = []

for station in raw_data:
    try:
        id = station["ID"]
        name = station["AddressInfo"]["Title"]
        lat = station["AddressInfo"]["Latitude"]
        lon = station["AddressInfo"]["Longitude"]
        conn = station["Connections"]
    except Exception as e:
        continue

    if station["OperatorID"] in operators_dict:
        operator = operators_dict[station["OperatorID"]]
    else:
        operator = "Unknown"

    data_to_insert.append({
        "name": name,
        "lat": lat,
        "lon": lon,
        "operator": operator,
        "connections": conn,
    })

with open('stations_data_to_insert.json', 'w', encoding='utf-8') as f:
    json.dump(data_to_insert, f, indent=4, ensure_ascii=False)

len(data_to_insert)

19716

In [24]:
st_example = clean_data.iloc[0]

In [25]:
pd.DataFrame(st_example)

,0
ID,479753
OperatorInfo,{'WebsiteURL': 'https://www.iberdrola.es/clien...
StatusType,"{'IsOperational': True, 'IsUserSelectable': Tr..."
UsageCost,"0,45€/kWh"
AddressInfo,"{'ID': 480142, 'Title': 'Parking Puerto Portal..."
Connections,"[{'ID': 805482, 'ConnectionTypeID': 33, 'Conne..."
NumberOfPoints,4.0


In [27]:
charg = pd.DataFrame(st_example["Connections"])

In [28]:
charg

,ID,ConnectionTypeID,ConnectionType,Reference,StatusTypeID,StatusType,LevelID,Level,Amps,Voltage,PowerKW,CurrentTypeID,CurrentType,Quantity,Comments
0,805482,33,"{'FormalName': 'IEC 62196-3 Configuration FF',...",None,50,"{'IsOperational': True, 'IsUserSelectable': Tr...",3,"{'Comments': '40KW and Higher', 'IsFastChargeC...",150,1000,60,30,"{'Description': 'Direct Current', 'ID': 30, 'T...",4,None


# DataBase Population

In [3]:
import json
import pandas as pd

In [4]:
import sys
sys.path.append('../..')

from src.Persistance.db_broker import DBBroker

In [5]:
with open('stations_data_to_insert.json', 'r') as f:
    data_to_insert = json.load(f)

len(data_to_insert)

19716

In [6]:
data_to_insert[0]

{'name': 'Parking Puerto Portals',
 'lat': 39.532410514731566,
 'lon': 2.570002495441713,
 'operator': 'Iberdrola',
 'connections': [{'ID': 805482,
   'ConnectionTypeID': 33,
   'ConnectionType': {'FormalName': 'IEC 62196-3 Configuration FF',
    'IsDiscontinued': False,
    'IsObsolete': False,
    'ID': 33,
    'Title': 'CCS (Type 2)'},
   'Reference': None,
   'StatusTypeID': 50,
   'StatusType': {'IsOperational': True,
    'IsUserSelectable': True,
    'ID': 50,
    'Title': 'Operational'},
   'LevelID': 3,
   'Level': {'Comments': '40KW and Higher',
    'IsFastChargeCapable': True,
    'ID': 3,
    'Title': 'Level 3:  High (Over 40kW)'},
   'Amps': 150,
   'Voltage': 1000,
   'PowerKW': 60,
   'CurrentTypeID': 30,
   'CurrentType': {'Description': 'Direct Current', 'ID': 30, 'Title': 'DC'},
   'Quantity': 4,
   'Comments': None}]}

## Stations

In [ ]:
query_station = """
    INSERT INTO charging_stations (station_name, operator, location)
    VALUES (?, ?, ST_GeomFromText(?))
"""

In [ ]:
batch_size = 10000

for i in range(0, len(data_to_insert), batch_size):
    batch = [
        (item["name"], item["operator"], f'POINT({item["lon"]} {item["lat"]})')
        for item in data_to_insert[i:i+batch_size]
    ]
    DBBroker().execute_many(query_station, batch)

2026-03-05 19:48:44,165 - DBBroker - INFO - Attempting to initialize Database Connection Pool...
2026-03-05 19:48:44,168 - DBBroker - INFO - Database connection pool established successfully.
2026-03-05 19:48:44,168 - DBBroker - DEBUG - Connection used from pool.
2026-03-05 19:48:44,169 - DBBroker - DEBUG - Executing batch WRITE query: 
    INSERT INTO charging_stations (station_name, operator, location)
    VALUES (?, ?, ST_GeomFromText(?))
 | Batch size: 10000
2026-03-05 19:48:44,333 - DBBroker - DEBUG - Batch write operation successful. Total rows affected: 0
2026-03-05 19:48:44,334 - DBBroker - DEBUG - Connection returned to pool.
2026-03-05 19:48:44,348 - DBBroker - DEBUG - Connection used from pool.
2026-03-05 19:48:44,348 - DBBroker - DEBUG - Executing batch WRITE query: 
    INSERT INTO charging_stations (station_name, operator, location)
    VALUES (?, ?, ST_GeomFromText(?))
 | Batch size: 9716
2026-03-05 19:48:44,427 - DBBroker - DEBUG - Batch write operation successful. Tota

## Chargers

In [7]:
st_example = data_to_insert[0]
charg = pd.DataFrame(st_example["connections"])
charg

,ID,ConnectionTypeID,ConnectionType,Reference,StatusTypeID,StatusType,LevelID,Level,Amps,Voltage,PowerKW,CurrentTypeID,CurrentType,Quantity,Comments
0,805482,33,"{'FormalName': 'IEC 62196-3 Configuration FF',...",None,50,"{'IsOperational': True, 'IsUserSelectable': Tr...",3,"{'Comments': '40KW and Higher', 'IsFastChargeC...",150,1000,60,30,"{'Description': 'Direct Current', 'ID': 30, 'T...",4,None


In [9]:
added = set()
types = list()
all_chargers = list()

for station in data_to_insert:
    for c in station["connections"]:
        all_chargers.append(c)
        if c["ConnectionTypeID"] not in added:
            print(c)
            types.append(c)
            added.add(c["ConnectionTypeID"])

len(types)

{'ID': 805482, 'ConnectionTypeID': 33, 'ConnectionType': {'FormalName': 'IEC 62196-3 Configuration FF', 'IsDiscontinued': False, 'IsObsolete': False, 'ID': 33, 'Title': 'CCS (Type 2)'}, 'Reference': None, 'StatusTypeID': 50, 'StatusType': {'IsOperational': True, 'IsUserSelectable': True, 'ID': 50, 'Title': 'Operational'}, 'LevelID': 3, 'Level': {'Comments': '40KW and Higher', 'IsFastChargeCapable': True, 'ID': 3, 'Title': 'Level 3:  High (Over 40kW)'}, 'Amps': 150, 'Voltage': 1000, 'PowerKW': 60, 'CurrentTypeID': 30, 'CurrentType': {'Description': 'Direct Current', 'ID': 30, 'Title': 'DC'}, 'Quantity': 4, 'Comments': None}
{'ID': 805260, 'ConnectionTypeID': 25, 'ConnectionType': {'FormalName': 'IEC 62196-2 Type 2', 'IsDiscontinued': False, 'IsObsolete': False, 'ID': 25, 'Title': 'Type 2 (Socket Only)'}, 'Reference': None, 'StatusTypeID': 50, 'StatusType': {'IsOperational': True, 'IsUserSelectable': True, 'ID': 50, 'Title': 'Operational'}, 'LevelID': 2, 'Level': {'Comments': 'Over 2 kW,

16

In [10]:
all_chargers_df = pd.DataFrame(all_chargers)
all_chargers_df["ConnectionTypeID"].value_counts()

ConnectionTypeID
25      12775
33       7901
2        3932
1036     3745
28       1896
30        375
16         80
17         77
1          24
0          14
13          8
29          4
18          3
4           2
32          1
27          1
Name: count, dtype: int64

In [ ]:
all_chargers_df = all_chargers_df[["ID", "ConnectionTypeID", "ConnectionType", "StatusTypeID", ]]

In [7]:
df_types = pd.DataFrame(types)
df_types = df_types[["ConnectionTypeID", "ConnectionType", "LevelID", "Level"]]

df_types

,ConnectionTypeID,ConnectionType,LevelID,Level
0,33,"{'FormalName': 'IEC 62196-3 Configuration FF',...",3.0,"{'Comments': '40KW and Higher', 'IsFastChargeC..."
1,25,"{'FormalName': 'IEC 62196-2 Type 2', 'IsDiscon...",2.0,"{'Comments': 'Over 2 kW, usually non-domestic ..."
2,28,"{'FormalName': 'CEE 7/4', 'IsDiscontinued': Fa...",2.0,"{'Comments': 'Over 2 kW, usually non-domestic ..."
3,2,"{'FormalName': 'IEC 62196-3 Configuration AA',...",3.0,"{'Comments': '40KW and Higher', 'IsFastChargeC..."
4,1036,"{'FormalName': 'IEC 62196-2', 'IsDiscontinued'...",2.0,"{'Comments': 'Over 2 kW, usually non-domestic ..."
5,0,"{'FormalName': 'Not Specified', 'IsDiscontinue...",NaN,None
6,1,"{'FormalName': 'SAE J1772-2009', 'IsDiscontinu...",2.0,"{'Comments': 'Over 2 kW, usually non-domestic ..."
7,17,"{'FormalName': None, 'IsDiscontinued': False, ...",2.0,"{'Comments': 'Over 2 kW, usually non-domestic ..."
8,30,"{'FormalName': None, 'IsDiscontinued': False, ...",2.0,"{'Comments': 'Over 2 kW, usually non-domestic ..."
9,16,"{'FormalName': None, 'IsDiscontinued': False, ...",2.0,"{'Comments': 'Over 2 kW, usually non-domestic ..."


In [56]:
a = df_types["ConnectionType"]

In [59]:
a = pd.DataFrame(a.iloc[:, 0].tolist())
a

,FormalName,IsDiscontinued,IsObsolete,ID,Title
0,IEC 62196-3 Configuration FF,False,False,33,CCS (Type 2)
1,IEC 62196-2 Type 2,False,False,25,Type 2 (Socket Only)
2,CEE 7/4,False,False,28,CEE 7/4 - Schuko - Type F
3,IEC 62196-3 Configuration AA,None,None,2,CHAdeMO
4,IEC 62196-2,False,False,1036,Type 2 (Tethered Connector)
5,Not Specified,None,None,0,Unknown
6,SAE J1772-2009,None,None,1,Type 1 (J1772)
7,NaN,False,False,17,CEE 5 Pin
8,NaN,False,False,30,Tesla (Model S/X)
9,NaN,False,False,16,CEE 3 Pin
